src/org/labkey/test/components/DomainDesignerPage.java

```diff
@@ -48,7 +54,10 @@ protected ElementCache newElementCache()
 
     protected class ElementCache extends LabKeyPage.ElementCache
     {
-        DomainFormPanel domainFormPanel(String domainName)
+        DomainFormPanel firstDomainFormPanel = new DomainFormPanel.DomainFormPanelFinder(getDriver())   // for situations where there's only one on the page
+                .findWhenNeeded(this);                                                          // and the caller is too lazy to specify which one they want
+
+        DomainFormPanel domainFormPanel(String domainName)                                              // for situations with multiple domainformpanels on the same page
```

Comment:
We will want different page classes for different designer contexts. e.g. `AssayDesignerPage` knows which domain designer panels it contains

Reply:
I like it.